plugins/node/instrumentation-mongoose/src/utils.ts

```diff
@@ -99,15 +99,27 @@ export function handleCallbackResponse(
   originalThis: any,
   span: Span,
   responseHook?: MongooseResponseCustomAttributesFunction,
-  moduleVersion: string | undefined = undefined
+  moduleVersion: string | undefined = undefined,
+  ...args: IArguments[]
 ) {
-  return exec.apply(originalThis, [
-    (err: Error, response: any) => {
-      err
-        ? setErrorStatus(span, err)
-        : applyResponseHook(span, response, responseHook, moduleVersion);
-      span.end();
-      return callback!(err, response);
-    },
-  ]);
+  const newArgs = [];
+  for (const [, argValue] of Object.entries(args[0])) {
+    newArgs.push(argValue);
+  }
+
+  let callbackArgumentIndex = 0;
+  if (newArgs.length === 2) {
+    callbackArgumentIndex = 1;
+  }
+
+  newArgs[callbackArgumentIndex] = (err: Error, response: any): any => {
```

Comment:
are we overwriting a user callback which is now not invoked? should we store it's value before overwriting and invoke it if present from our response callback?

Reply:
Never mind, I was thinking that the `callback` is already part of the args and we can potentially extract it from it instead of sending it with the function. But now that I think about it again, I think it is more readable the way it is now 👍